### PR TITLE
[BugFix] DeepSeek-OCR: apply NoRepeatNGramLogitsProcessor to greedy path

### PR DESCRIPTION
**Pull request by YuanpingSong**
## Purpose
* What: Fix DeepSeek-OCR custom logits processor classification so it runs for greedy decoding. Specifically, NGramPerReqLogitsProcessor.is_argmax_invariant() is flipped to `False` in [vllm/model_executor/models/deepseek_ocr.py](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/models/deepseek_ocr.py#L163).

* Why: In the v1 sampler, argmax-invariant processors are applied only in the random/temperature path. With temperature=0.0 (greedy), the sampler [returns early](https://github.com/vllm-project/vllm/blob/main/vllm/v1/sample/sampler.py#L162-L169) and skips argmax-invariant processors. Since the n‑gram processor bans tokens by setting logits to -inf (which can change the argmax), it must be non‑argmax‑invariant to affect greedy decoding. This change restores expected repeat suppression for OCR use cases that require deterministic decoding.

## Test Plan

1. serve:

```
vllm serve deepseek-ai/DeepSeek-OCR --no-enable-prefix-caching --mm-processor-cache-gb 0 --gpu-memory-utilization 0.9 --max_num_batched_tokens 8192 --enable-log-requests --logits-processors vllm.model_executor.models.deepseek_ocr:NGramPerReqLogitsProcessor --mm-processor-kwargs '{"base_size":1024,"image_size":640,"crop_mode": true}'

```

2. Request:

```

resp = client.chat.completions.create(
    model='deepseek-ai/DeepSeek-OCR',
    messages=[{
    'role': 'user',
    'content': [
        {'type': 'image_url', 'image_url': {'url': b64_png(TEST_IMAGE_PATH)}},
        {'type': 'text', 'text': PROMPT},
    ],
    }],
    max_tokens=4096,
    temperature=0,
    extra_body={
    'skip_special_tokens': False,
    'vllm_xargs': {
        'ngram_size': 2,
        'window_size': 100,
        'whitelist_token_ids': [128821, 128822],
    },
},
)
```

## Test Result

Before patch: `NGramPerReqLogitsProcessor`  not called

After patch: `banned_tokens` being populated in [NoRepeatNGramLogitsProcessor](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/models/deepseek_ocr.py#L116)

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Review by Isotr0py — APPROVED** [review]
Thanks for fixing!
